e2e-tests/tests/dynamicCluster.spec.ts

```diff
@@ -0,0 +1,171 @@
+import { test, expect } from "@playwright/test";
+import { HeadlampPage } from './headlampPage';
+const yaml = require('yaml');
+const fs = require('fs').promises;
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+test('Store modified kubeconfig to IndexDB and check if present', async ({ page }) => {
+  const headlampPage = new HeadlampPage(page);
+
+  await headlampPage.authenticate();
+  await page.waitForLoadState('load');
+
+  const base64EncodedKubeconfig = await getBase64EncodedKubeconfig(page);
+  await saveKubeconfigToIndexDB(page, base64EncodedKubeconfig);
+  await page.waitForLoadState('load');
+
+  const storedKubeconfig = await getKubeconfigFromIndexDB(page);
+  await page.waitForLoadState('load');
+
+  expect(storedKubeconfig).not.toBeNull();
+});
+
+test('check test is present in cluster and working', async ({ page }) => {
+  const headlampPage = new HeadlampPage(page);
+
+  await headlampPage.authenticate();
+  await page.waitForLoadState('load');
+
+  const base64EncodedKubeconfig = await getBase64EncodedKubeconfig(page);
+  await saveKubeconfigToIndexDB(page, base64EncodedKubeconfig);
+  await page.waitForLoadState('load');
+
+  const storedKubeconfig = await getKubeconfigFromIndexDB(page);
+  await page.waitForLoadState('load');
+
+  expect(storedKubeconfig).not.toBeNull();
+
+  await headlampPage.navigateTopage('/c/test');
+  await headlampPage.pageLocatorContent('h2:has-text("Overview")', 'Overview');
+});
+
+const getBase64EncodedKubeconfig = async (page) => {
+  // Use kubectl command-line tool to get the kubeconfig
+  const { stdout, stderr } = await exec('kubectl config view --output json');
+  if (stderr) {
+    console.error('Error fetching Minikube kubeconfig:', stderr);
+    return;
+  }
+
+  // Parse the kubeconfig JSON
+  const kubeconfig = JSON.parse(stdout);
+  // Update the existing cluster and context names to "test"
+  kubeconfig.clusters[0].name = 'test';
+  kubeconfig.clusters[0].cluster.server = 'https://10.96.0.1:443';
```

Comment:
the server url is hard coded here, can fail if minikube starts in another address due to some reason.

Reply:
It won't since it is default in minikube if not changes. The `10.96.0.0/12`: Used by service cluster IP’s and the first service that is created is that of minikube when it bootstraps the cluster, so it won't be a problem. https://minikube.sigs.k8s.io/docs/handbook/vpn_and_proxy/